traffic_ops/traffic_ops_golang/acme/acme_account.go

```diff
@@ -0,0 +1,203 @@
+		return
+	}
+
+	var prevAccount tc.AcmeAccount
+	err := tx.QueryRow("SELECT email, private_key, uri, provider from acme_account where email = $1 and provider = $2", acmeAccount.Email, acmeAccount.Provider).Scan(&prevAccount.Email, &prevAccount.PrivateKey, &prevAccount.Uri, &prevAccount.Provider)
+	if err == sql.ErrNoRows {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, errors.New(fmt.Sprintf("acme account not found")), nil)
+		return
+	}
+	if err != nil {
+		api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, errors.New(fmt.Sprintf("checking if acme account with email %s and provider %s exists: %v", *acmeAccount.Email, *acmeAccount.Provider, err.Error())))
+		return
+	}
+
+	resultRows, err := inf.Tx.NamedQuery(updateQuery, acmeAccount)
+	if err != nil {
+		userErr, sysErr, errCode := api.ParseDBError(err)
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer resultRows.Close()
+
+	rowsAffected := 0
+	for resultRows.Next() {
+		rowsAffected++
+	}
+	if rowsAffected == 0 {
+		api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, errors.New("acme account update: no account was updated"))
+		return
+	}
+
+	alerts := tc.CreateAlerts(tc.SuccessLevel, "Acme account updated")
+	api.WriteAlertsObj(w, r, http.StatusCreated, alerts, acmeAccount)
+
+	changeLogMsg := fmt.Sprintf("ACME ACCOUNT: %s %s, ACTION: updated", *acmeAccount.Email, *acmeAccount.Provider)
+	api.CreateChangeLogRawTx(api.ApiChange, changeLogMsg, inf.User, tx)
+}
+
+func Delete(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, []string{"provider", "email"}, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	provider := inf.Params["provider"]
+	email := inf.Params["email"]
+
+	tx := inf.Tx.Tx
+
+	var prevAccount tc.AcmeAccount
+	err := tx.QueryRow("SELECT email, private_key, uri, provider from acme_account where email = $1 and provider = $2", email, provider).Scan(&prevAccount.Email, &prevAccount.PrivateKey, &prevAccount.Uri, &prevAccount.Provider)
+	if err == sql.ErrNoRows {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, errors.New(fmt.Sprintf("acme account not found")), nil)
+		return
+	}
+	if err != nil {
+		api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, errors.New(fmt.Sprintf("checking if acme account with email %s and provider %s exists: %v", email, provider, err.Error())))
+		return
+	}
+
+	if _, err := tx.Exec(deleteQuery, email, provider); err != nil {
+		api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, errors.New(fmt.Sprintf("deleting acme account with email %s and provider %s: %v", email, provider, err.Error())))
+		return
+	}
+
+	api.WriteRespAlert(w, r, tc.SuccessLevel, "Acme account deleted")
+
+	changeLogMsg := fmt.Sprintf("ACME ACCOUNT: %s %s, ACTION: deleted", email, provider)
+	api.CreateChangeLogRawTx(api.ApiChange, changeLogMsg, inf.User, tx)
+}
```

Comment:
Do we want to ensure files end with a newline? Including the `goose` SQL file as well?

Reply:
Doesn't `go fmt` do that?

Reply:
i think they are there already. i have extra lines locally but idk why it doesnt show them in the file diff on here

Reply:
Github will show a :no_entry_sign: at the end of a file that doesn't have a terminating newline. Text editors treat tend to treat terminating newlines as a "blank line" at the end of a file, but really a newline character terminates a line and so belongs to the final non-"blank" line, which is how GitHub handles it.